File: backend/app/schemas/cloud.py
from datetime import datetime

from pydantic import BaseModel


class RegionResponse(BaseModel):
    id: str
    name: str
    code: str
    is_primary: bool
    status: str

    class Config:
        from_attributes = True

class ClusterResponse(BaseModel):
    id: str
    name: str
    region_id: str
    provider: str
    nodes_count: int
    k8s_version: str
    status: str

    class Config:
        from_attributes = True

class DeploymentResponse(BaseModel):
    id: str
    release_version: str
    strategy: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class BackupResponse(BaseModel):
    id: str
    workspace_id: str
    backup_type: str
    size_bytes: int
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class FeatureFlagResponse(BaseModel):
    id: str
    flag_key: str
    is_enabled: bool
    description: str
    percentage_rollout: int

    class Config:
        from_attributes = True

class CloudMetricsResponse(BaseModel):
    total_clusters: int
    total_nodes: int
    global_traffic_rps: int
    avg_region_latency_ms: int
    cpu_utilization_pct: float
    memory_utilization_pct: float
    active_canary_deployments: int

File: frontend/src/modules/cloud/types/cloud.ts
export interface RegionRecord {
  id: string;
  name: string;
  code: string;
  isPrimary: boolean;
  status: 'healthy' | 'degraded' | 'maintenance';
}

export interface ClusterRecord {
  id: string;
  name: string;
  regionId: string;
  provider: string;
  nodesCount: number;
  k8sVersion: string;
  status: 'active' | 'draining' | 'failed';
}

export interface DeploymentRecord {
  id: string;
  releaseVersion: string;
  strategy: string;
  status: 'in_progress' | 'succeeded' | 'rolled_back';
  createdAt: string;
}

export interface BackupRecord {
  id: string;
  workspaceId: string;
  backupType: string;
  sizeBytes: number;
  status: 'completed' | 'failed' | 'in_progress';
  createdAt: string;
}

export interface FeatureFlagRecord {
  id: string;
  flagKey: string;
  isEnabled: boolean;
  description: string;
  percentageRollout: number;
}

export interface CloudMetricsRecord {
  totalClusters: number;
  totalNodes: number;
  globalTrafficRps: number;
  avgRegionLatencyMs: number;
  cpuUtilizationPct: number;
  memoryUtilizationPct: number;
  activeCanaryDeployments: number;
}
